pdns/syncres.cc

```diff
@@ -1074,40 +1074,64 @@ vector<ComboAddress> SyncRes::getAddrs(const DNSName &qname, unsigned int depth,
   d_followCNAME = true;
 
   try {
-    vState newState = vState::Indeterminate;
-    res_t resv4;
-    // If IPv4 ever becomes second class, we should revisit this
-    if (s_doIPv4 && doResolve(qname, QType::A, resv4, depth+1, beenthere, newState) == 0) {  // this consults cache, OR goes out
-      for (auto const &i : resv4) {
-        if (i.d_type == QType::A) {
+    // First look for both A and AAAA in the cache and be satisfied if we find anything
+    res_t cset;
+    if (s_doIPv4 && g_recCache->get(d_now.tv_sec, qname, QType::A, false, &cset, d_cacheRemote, d_refresh, d_routingTag) > 0) {
+      for (const auto &i : cset) {
+        if (i.d_ttl > (unsigned int)d_now.tv_sec ) {
```

Comment:
I don't think we need that test, `MemRecursorCache::get` should only return a positive value is the TTD is still good?

Reply:
Right, this was a c/p from the existing cide. I'll double check and adapt.

Reply:
The return valiue of `get()` reflects the ttd of the cache entry, which should be <= the minimum of the ttd of the individual records in the cache entry. So I think the condition can be removed indeed.